### PR TITLE
Add initial strategies document

### PR DESCRIPTION
**Pull request by k0da**
* This moves strategy options into a separate document page for better
  visibility.
* Describes new geoip strategy option.

Signed-off-by: Dinar Valeev <dinar.valeev@absa.africa>

### CODE REVIEW

**Review by kuritka — COMMENTED** [review]
Hi, this comment is not only about GeoIP, but overall strategies. Shouldn't we provide more detailed examples ? (e.g: some pictures + example configurations below). Make it more user friendly....

**Comment by k0da** [comment]
@kuritka my intent is to provide an overview what is the difference between them. How to configure spec and architecture is available already.

**Review by kuritka — APPROVED** [review]

File: docs/index.md [hunk 1]
```diff
@@ -143,10 +143,7 @@ If the Pods in cluster **Y** were to once again become healthy (liveness and rea
 
 The following load balancing strategies, as it relates to resolving Ingress node IPs, should be provided as part of the initial implementation:
```
Comment (ytsarev) [inline docs/index.md:144]: Colon(:) look alien after modifications

File: docs/index.md [hunk 2]
```diff
@@ -143,10 +143,7 @@ If the Pods in cluster **Y** were to once again become healthy (liveness and rea
...
 
-* **Round robin** - The default strategy
-* **Weighted round robin** - Specialisation of the above strategy but where a percentage weighting is applied to determine which cluster's Ingress node IPs to resolve. E.g. 80% cluster **X** and 20% cluster **Y**
-* **Failover** - Pinned to a specified primary cluster until that cluster has no available Pods, upon which the next available cluster's Ingress node IPs will be resolved. When Pods are again available on the primary cluster, the primary cluster will once again be the only eligible cluster for which cluster Ingress node IPs will be resolved
-* **Manual** - Eligibility is manually specified as to which cluster(s) are eligible. If there are no available Pods in the specified clusters, then no cluster Ingress node IPs will be resolved and the client will get a [`NXDOMAIN`](https://www.dnsknowledge.com/whatis/nxdomain-non-existent-domain-2/) response
+see [strategies](strategy.md) for available options.
```
Comment (ytsarev) [inline docs/index.md:146]: I would still list the strategy names in the index at least

File: docs/strategy.md [hunk 3]
```diff
@@ -0,0 +1,9 @@
+# k8gb strategy options
+## failover
+Pinned to a specified primary cluster until that cluster has no available Pods, upon which the next available cluster's Ingress node IPs will be resolved. When Pods are again available on the primary cluster, the primary cluster will once again be the only eligible cluster for which cluster Ingress node IPs will be resolved
+
+## roundRobin (*default*)
+Returns both cluster endpoints in round robin matter.
```
Comment (ytsarev) [inline docs/strategy.md:6]:
```suggestion
Returns both cluster endpoints in round-robin manner.
```

File: docs/strategy.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+# k8gb strategy options
+## failover
+Pinned to a specified primary cluster until that cluster has no available Pods, upon which the next available cluster's Ingress node IPs will be resolved. When Pods are again available on the primary cluster, the primary cluster will once again be the only eligible cluster for which cluster Ingress node IPs will be resolved
```
Comment (ytsarev) [inline docs/strategy.md:3]:
```suggestion
Pinned to a specified primary cluster until workload on that cluster has no available Pods, upon which the next available cluster's Ingress node IPs will be resolved. When Pods are again available on the primary cluster, the primary cluster will once again be the only eligible cluster for which cluster Ingress node IPs will be resolved
```

File: docs/strategy.md [hunk 5]
```diff
@@ -0,0 +1,9 @@
...
+
+## roundRobin (*default*)
+Returns both cluster endpoints in round robin matter.
+
+## geoip
+Similar to `failover` mode, but returns "closest" cluster to the client initiating request. This requires specially crafted GeoIP database (see [this](https://github.com/k8gb-io/coredns-crd-plugin/tree/main/terratest/geogen) for example) and DNS resolver to support EDNS0 extension (CLIENT-SUBNET in particular). If client subnet is not in GeoIP Database, all available endpoints are returned
```
Comment (ytsarev) [inline docs/strategy.md:9]:
```suggestion
Similar to `failover` mode, but returns "closest" cluster to the client initiating request. This requires a specially crafted GeoIP database (see [this](https://github.com/k8gb-io/coredns-crd-plugin/tree/main/terratest/geogen) for example) and DNS resolver to support EDNS0 extension (CLIENT-SUBNET in particular). If the client subnet is not in GeoIP Database, all available endpoints are returned
```

**Review by ytsarev — CHANGES_REQUESTED** [review]

**Comment by k0da** [comment]
@ytsarev amended

File: docs/index.md [hunk 6]
```diff
@@ -142,11 +142,11 @@ If the Pods in cluster **Y** were to once again become healthy (liveness and rea
 ## Load balancing strategies
 
 The following load balancing strategies, as it relates to resolving Ingress node IPs, should be provided as part of the initial implementation:
+* **Round robin**
+* **Failover**
+* **GeoIP**
```
Comment (ytsarev) [inline docs/index.md:147]: Please make the case consistent with the config and the strategy doc, e.g.
```suggestion
* **geoip**
```

**Comment by k0da** [comment]
@ytsarev proposal commited, thanks

File: docs/index.md [hunk 7]
```diff
@@ -142,11 +142,11 @@ If the Pods in cluster **Y** were to once again become healthy (liveness and rea
 ## Load balancing strategies
 
 The following load balancing strategies, as it relates to resolving Ingress node IPs, should be provided as part of the initial implementation:
+* **Round robin**
```
Comment (ytsarev) [inline docs/index.md:145]:
```suggestion
* **roundRobin**
```

File: docs/index.md [hunk 8]
```diff
@@ -142,11 +142,11 @@ If the Pods in cluster **Y** were to once again become healthy (liveness and rea
...
+* **Failover**
```
Comment (ytsarev) [inline docs/index.md:146]:
```suggestion
* **failover**
```

File: docs/strategy.md [hunk 9]
```diff
@@ -0,0 +1,9 @@
+# k8gb strategy options
+## failover
+Pinned to a specified primary cluster until workload on that cluster has no available Pods, upon which the next available cluster's Ingress node IPs will be resolved. When Pods are again available on the primary cluster, the primary cluster will once again be the only eligible cluster for which cluster Ingress node IPs will be resolved
+
+## roundRobin (*default*)
```
Comment (ytsarev) [inline docs/strategy.md:5]: Please place this strategy first on the list. What do you mean by *default* ?

**Review by ytsarev — CHANGES_REQUESTED** [review]
Sorry for not being clear enough, I meant to use proper case for all the strategies - provided additional suggestions

File: docs/strategy.md (hunk 9 above)
Comment (k0da) [reply docs/strategy.md:5]: @ytsarev default means this is default strategy

Comment (ytsarev) [reply docs/strategy.md:5]: @k0da yeah. But why is it default? we always specify the strategies explicitly

Comment (k0da) [reply docs/strategy.md:5]: @ytsarev inspired by docs I'm changing:
`* **Round robin** - The default strategy`

Comment (ytsarev) [reply docs/strategy.md:5]: @k0da still I am challenging it :) It's not real default one :)

Comment (k0da) [reply docs/strategy.md]: @ytsarev as you wish, amended

File: docs/strategy.md [hunk 10]
```diff
@@ -0,0 +1,9 @@
+# k8gb strategy options
+## failover
+Pinned to a specified primary cluster until workload on that cluster has no available Pods, upon which the next available cluster's Ingress node IPs will be resolved. When Pods are again available on the primary cluster, the primary cluster will once again be the only eligible cluster for which cluster Ingress node IPs will be resolved
+
+## roundRobin
```
Comment (ytsarev) [inline docs/strategy.md:5]: Please move it to the first place to be consistent with https://github.com/k8gb-io/k8gb/pull/556/files#diff-b4d68dc855d0f9476d3f2ee343853bd21bf82ea9960d0cf06661baa244439dd6R145

**Review by ytsarev — CHANGES_REQUESTED** [review]

File: docs/strategy.md (hunk 10 above)
Comment (k0da) [reply docs/strategy.md:2]: It was lost during rebase, now it is there

**Review by ytsarev — APPROVED** [review]
lgtm

**Pull request merged**